src/controllers/project.js

```diff
@@ -0,0 +1,92 @@
+const Project = require('../models/project');
+
+exports.getAllProjects = async (_, res) => {
+  try {
+    const projects = await Project.find();
+    res.json(projects);
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+  }
+};
+
+exports.getOneProject = async (req, res) => {
+  const { id } = req.params;
+  try {
+    const project = await Project.findById(id);
+    if (!project) {
+      res
+        .status(422)
+        .json({ message: "the project you are looking for wasn't found" });
+    } else {
+      res.json(project);
+    }
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+  }
+};
+
+exports.addProject = async (req, res) => {
+  const projectData = req.body;
+  try {
+    const newProject = await Project.create(projectData);
+    res.status(201).json(newProject);
+    return projectData;
+  } catch (err) {
+    res.status(422).json({ message: err.message });
+    return err.message;
+  }
+};
+
+exports.filterProjects = async (req, res) => {
+  const { creator, address } = req.query;
+  if (!creator && !address) {
+    res
+      .status(400)
+      .json({ message: 'make sure you send a valid query parameter' });
+  } else {
+    const query = {};
+    if (creator) query.creator = creator;
+    if (address) query.address = address;
+    try {
+      const projects = await Project.find(query);
+      res.json(projects);
+    } catch (err) {
+      res.status(422).json({ message: err.message });
+    }
+  }
+};
+
+exports.updateProject = async (req, res) => {
+  const { id } = req.params;
+  try {
+    const updatedProject = await Project.findByIdAndUpdate(id, req.body, {
+      new: true,
+    });
```

Comment:
I think you should use `$set` in the update query, so that only the fields in the request body are updated. Otherwise mongoose changes the whole document to what is in the request body.

Reply:
Hi @Shrreya. I did this according to: [_all top level update keys which are not atomic operation names are treated as set operations_](https://mongoosejs.com/docs/api/model.html#model_Model.findByIdAndUpdate)